testsuite/tools/expect_test.ml

```diff
@@ -328,13 +328,70 @@ let main fname =
   process_expect_file fname;
   exit 0
 
+module Options = Main_args.Make_bytetop_options (struct
+  let set r () = r := true
+  let clear r () = r := false
+  open Clflags
+  let _absname = set Location.absname
+  let _I dir =
+    let dir = Misc.expand_directory Config.standard_library dir in
+    include_dirs := dir :: !include_dirs
+  let _init s = init_file := Some s
+  let _noinit = set noinit
+  let _labels = clear classic
+  let _alias_deps = clear transparent_modules
+  let _no_alias_deps = set transparent_modules
+  let _app_funct = set applicative_functors
+  let _no_app_funct = clear applicative_functors
+  let _noassert = set noassert
+  let _nolabels = set classic
+  let _noprompt = set noprompt
+  let _nopromptcont = set nopromptcont
+  let _nostdlib = set no_std_include
+  let _open s = open_modules := s :: !open_modules
+  let _ppx _s = (* disabled *) ()
+  let _principal = set principal
+  let _no_principal = clear principal
+  let _rectypes = set recursive_types
+  let _no_rectypes = clear recursive_types
+  let _safe_string = clear unsafe_string
+  let _short_paths = clear real_paths
+  let _stdin () = (* disabled *) ()
+  let _strict_sequence = set strict_sequence
+  let _no_strict_sequence = clear strict_sequence
+  let _strict_formats = set strict_formats
+  let _no_strict_formats = clear strict_formats
+  let _unboxed_types = set unboxed_types
+  let _no_unboxed_types = clear unboxed_types
+  let _unsafe = set fast
+  let _unsafe_string = set unsafe_string
+  let _version () = (* disabled *) ()
+  let _vnum () = (* disabled *) ()
+  let _no_version = set noversion
+  let _w s = Warnings.parse_options false s
+  let _warn_error s = Warnings.parse_options true s
+  let _warn_help = Warnings.help_warnings
+  let _dparsetree = set dump_parsetree
+  let _dtypedtree = set dump_typedtree
+  let _dsource = set dump_source
+  let _drawlambda = set dump_rawlambda
+  let _dlambda = set dump_lambda
+  let _dflambda = set dump_flambda
+  let _dtimings () = profile_columns := [ `Time ]
+  let _dprofile () = profile_columns := Profile.all_columns
+  let _dinstr = set dump_instr
+
+  let _args = Arg.read_arg
+  let _args0 = Arg.read_arg0
+
+  let anonymous s = (*disabled*) ()
```

Comment:
Does `(* disabled *) ()` here have the effect that anonymous parameters are silently ignored? If that was the case, I think it would not be a very good choice.

Reply:
No, the comment is unclear; the anonymous arguments are handled by the `main` function in `expect_test`, this `anonymous` function is only here to fulfill the functor argument signature.

Reply:
If I understand the code correctly, passing the command-line arguments `-- foo` would still call the `anonymous` functor argument on `foo` and `bar`. (If there was no way to have this functor argument ever called at run time, this would be an API bug and we would consider removing it.)

Reply:
I stand corrected; `- -a.ml` will indeed call `anonymous "-a.ml"`. Fixed.